actors/builtin/miner/miner_actor.go

```diff
@@ -102,7 +102,25 @@ func (a Actor) Constructor(rt Runtime, params *ConstructorParams) *adt.EmptyValu
 	periodStart := nextProvingPeriodStart(currEpoch, offset)
 	Assert(periodStart > currEpoch)
 
-	state, err := ConstructState(emptyArray, emptyMap, emptyDeadlinesCid, owner, worker, params.PeerId, params.Multiaddrs, params.SealProofType, periodStart)
+	sectorSize, err := params.SealProofType.SectorSize()
+	builtin.RequireNoErr(rt, err, exitcode.ErrIllegalArgument, "no sector size for seal proof type %d: %w", params.SealProofType)
```

Comment:
Note: `RequireNoErr` does the wrapping, don't append `%w`. But push this back into a function anyway (see below)